die() call in iamcal/sql-parser crashes PHPStan

<!-- First, thank you for reporting a bug. That takes time and we appreciate that! -->

- Larastan Version: 3.2.0
- Laravel Version: 12.2.0

### Description
<!-- Please describe your problem here. -->
This was a nasty one. After some recent changes, PHPStan started crashing for no obvious reason:

```
 -- ---------------------------------------------------------------------------- 
     Error                                                                       
 -- ---------------------------------------------------------------------------- 
     Some parallel worker jobs have not finished. while running parallel worker  
 -- ---------------------------------------------------------------------------- 


                                                                                                                        
 [ERROR] Found 1 error                                                                                                  
                                                                                                                        

⚠️  Result is incomplete because of severe errors. ⚠️
   Fix these errors first and then re-run PHPStan
   to get all reported errors.
```

Running in debug mode got me at least a bit more info:

```
Unsupported field type: CHARACTER⏎
```

Searching for that message, I [found it](https://github.com/iamcal/SQLParser/blob/v0.5/src/SQLParser.php#L730) in the `parse_field` method in the [`iamcal/sql-parser` package](https://github.com/iamcal/SQLParser):

```php
die("Unsupported field type: {$f['type']}");
```

Throwing an exception instead at least bubbles the error up to the console with a proper stack trace, which you can find below:

```
 -- ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
     Error
 -- ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
     Internal error: Unsupported field type: CHARACTER while analysing file /app/Models/Search/SearchQueryQaRun.php
     Post the following stack trace to https://github.com/larastan/larastan/issues/new?template=bug-report.md:
     ## /vendor/iamcal/sql-parser/src/SQLParser.php(732)
     #0 /vendor/iamcal/sql-parser/src/SQLParser.php(549): iamcal\SQLParser->parse_field(Array)
     #1 /vendor/iamcal/sql-parser/src/SQLParser.php(345): iamcal\SQLParser->parse_field_or_key(Array, Array, Array)
     #2 /vendor/iamcal/sql-parser/src/SQLParser.php(301): iamcal\SQLParser->parse_create_definition(Array, 16)
     #3 /vendor/iamcal/sql-parser/src/SQLParser.php(216): iamcal\SQLParser->parse_create_table(Array, 16, 112)
     #4 /vendor/iamcal/sql-parser/src/SQLParser.php(28): iamcal\SQLParser->walk(Array, '--\n-- PostgreSQ...', Array)
     #5 /vendor/larastan/larastan/src/Properties/SquashedMigrationHelper.php(68): iamcal\SQLParser->parse('--\n-- PostgreSQ...')
     #6 /vendor/larastan/larastan/src/Properties/ModelPropertyHelper.php(227): Larastan\Larastan\Properties\SquashedMigrationHelper->initializeTables()
     #7 /vendor/larastan/larastan/src/Properties/ModelPropertyHelper.php(48): Larastan\Larastan\Properties\ModelPropertyHelper->loadMigrations()
     #8 /vendor/larastan/larastan/src/Properties/ModelPropertyExtension.php(25): Larastan\Larastan\Properties\ModelPropertyHelper->hasDatabaseProperty(Object(PHPStan\Reflection\ClassReflection), 'type')
     #9 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Reflection/ClassReflection.php(427): Larastan\Larastan\Properties\ModelPropertyExtension->hasProperty(Object(PHPStan\Reflection\ClassReflection), 'type')
     #10 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Type/ObjectType.php(141): PHPStan\Reflection\ClassReflection->hasProperty('type')
     #11 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php(3934): PHPStan\Type\ObjectType->hasProperty('type')
     #12 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php(3944): PHPStan\Analyser\MutatingScope->getPropertyReflection(Object(PHPStan\Type\ObjectType), 'type')
     #13 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php(1568): PHPStan\Analyser\MutatingScope->propertyFetchType(Object(PHPStan\Type\ObjectType), 'type', Object(PhpParser\Node\Expr\PropertyFetch))
     #14 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php(1574): PHPStan\Analyser\MutatingScope->{closure:PHPStan\Analyser\MutatingScope::resolveType():1567}()
     #15 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php(554): PHPStan\Analyser\MutatingScope->resolveType('$run->type', Object(PhpParser\Node\Expr\PropertyFetch))
     #16 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Dependency/DependencyResolver.php(239): PHPStan\Analyser\MutatingScope->getType(Object(PhpParser\Node\Expr\PropertyFetch))
     #17 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php(155): PHPStan\Dependency\DependencyResolver->resolveDependencies(Object(PhpParser\Node\Expr\PropertyFetch), Object(PHPStan\Analyser\MutatingScope))
     #18 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Node/ClassStatementsGatherer.php(116): PHPStan\Analyser\FileAnalyser->{closure:PHPStan\Analyser\FileAnalyser::analyseFile():85}(Object(PhpParser\Node\Expr\PropertyFetch), Object(PHPStan\Analyser\MutatingScope))
     #19 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(654): PHPStan\Node\ClassStatementsGatherer->__invoke(Object(PhpParser\Node\Expr\PropertyFetch), Object(PHPStan\Analyser\MutatingScope))
     #20 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(731): PHPStan\Analyser\NodeScopeResolver::{closure:PHPStan\Analyser\NodeScopeResolver::processStmtNode():653}(Object(PhpParser\Node\Expr\PropertyFetch), Object(PHPStan\Analyser\MutatingScope))
     #21 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(3944): PHPStan\Analyser\NodeScopeResolver::{closure:PHPStan\Analyser\NodeScopeResolver::processStmtNode():730}(Object(PhpParser\Node\Expr\PropertyFetch), Object(PHPStan\Analyser\MutatingScope))
     #22 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(1881): PHPStan\Analyser\NodeScopeResolver->processAssignVar(Object(PHPStan\Analyser\MutatingScope), Object(PhpParser\Node\Stmt\Expression), Object(PhpParser\Node\Expr\PropertyFetch), Object(PhpParser\Node\Expr\Variable), Object(Closure),
     Object(PHPStan\Analyser\ExpressionContext), Object(Closure), true)
     #23 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(730): PHPStan\Analyser\NodeScopeResolver->processExprNode(Object(PhpParser\Node\Stmt\Expression), Object(PhpParser\Node\Expr\Assign), Object(PHPStan\Analyser\MutatingScope), Object(Closure), Object(PHPStan\Analyser\ExpressionContext))
     #24 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(456): PHPStan\Analyser\NodeScopeResolver->processStmtNode(Object(PhpParser\Node\Stmt\Expression), Object(PHPStan\Analyser\MutatingScope), Object(Closure), Object(PHPStan\Analyser\StatementContext))
     #25 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(653): PHPStan\Analyser\NodeScopeResolver->processStmtNodes(Object(PhpParser\Node\Stmt\ClassMethod), Array, Object(PHPStan\Analyser\MutatingScope), Object(Closure), Object(PHPStan\Analyser\StatementContext))
     #26 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(456): PHPStan\Analyser\NodeScopeResolver->processStmtNode(Object(PhpParser\Node\Stmt\ClassMethod), Object(PHPStan\Analyser\MutatingScope), Object(PHPStan\Node\ClassStatementsGatherer), Object(PHPStan\Analyser\StatementContext))
     #27 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(793): PHPStan\Analyser\NodeScopeResolver->processStmtNodes(Object(PhpParser\Node\Stmt\Class_), Array, Object(PHPStan\Analyser\MutatingScope), Object(PHPStan\Node\ClassStatementsGatherer), Object(PHPStan\Analyser\StatementContext))
     #28 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(456): PHPStan\Analyser\NodeScopeResolver->processStmtNode(Object(PhpParser\Node\Stmt\Class_), Object(PHPStan\Analyser\MutatingScope), Object(Closure), Object(PHPStan\Analyser\StatementContext))
     #29 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(760): PHPStan\Analyser\NodeScopeResolver->processStmtNodes(Object(PhpParser\Node\Stmt\Namespace_), Array, Object(PHPStan\Analyser\MutatingScope), Object(Closure), Object(PHPStan\Analyser\StatementContext))
     #30 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(404): PHPStan\Analyser\NodeScopeResolver->processStmtNode(Object(PhpParser\Node\Stmt\Namespace_), Object(PHPStan\Analyser\MutatingScope), Object(Closure), Object(PHPStan\Analyser\StatementContext))
     #31 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php(172): PHPStan\Analyser\NodeScopeResolver->processNodes(Array, Object(PHPStan\Analyser\MutatingScope), Object(Closure))
     #32 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Command/WorkerCommand.php(136): PHPStan\Analyser\FileAnalyser->analyseFile('/Users/moritz/P...', Array, Object(PHPStan\Rules\LazyRegistry), Object(PHPStan\Collectors\Registry), NULL)
     #33 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/EventEmitterTrait.php(111): PHPStan\Command\WorkerCommand::{closure:PHPStan\Command\WorkerCommand::runWorker():117}(Array)
     #34 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/clue/ndjson-react/src/Decoder.php(117): _PHPStan_781aefaf6\Evenement\EventEmitter->emit('data', Array)
     #35 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/EventEmitterTrait.php(111): _PHPStan_781aefaf6\Clue\React\NDJson\Decoder->handleData(Array)
     #36 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/react/stream/src/Util.php(62): _PHPStan_781aefaf6\Evenement\EventEmitter->emit('data', Array)
     #37 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/EventEmitterTrait.php(111): _PHPStan_781aefaf6\React\Stream\Util::{closure:_PHPStan_781aefaf6\React\Stream\Util::forwardEvents():61}('{"action":"anal...')
     #38 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/react/stream/src/DuplexResourceStream.php(168): _PHPStan_781aefaf6\Evenement\EventEmitter->emit('data', Array)
     #39 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/react/event-loop/src/StreamSelectLoop.php(201): _PHPStan_781aefaf6\React\Stream\DuplexResourceStream->handleData(Resource id #9515)
     #40 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/react/event-loop/src/StreamSelectLoop.php(173): _PHPStan_781aefaf6\React\EventLoop\StreamSelectLoop->waitForStreamActivity(NULL)
     #41 phar:///vendor/phpstan/phpstan/phpstan.phar/src/Command/WorkerCommand.php(96): _PHPStan_781aefaf6\React\EventLoop\StreamSelectLoop->run()
     #42 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Command/Command.php(259): PHPStan\Command\WorkerCommand->execute(Object(_PHPStan_781aefaf6\Symfony\Component\Console\Input\ArgvInput), Object(_PHPStan_781aefaf6\Symfony\Component\Console\Output\ConsoleOutput))
     #43 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(870): _PHPStan_781aefaf6\Symfony\Component\Console\Command\Command->run(Object(_PHPStan_781aefaf6\Symfony\Component\Console\Input\ArgvInput), Object(_PHPStan_781aefaf6\Symfony\Component\Console\Output\ConsoleOutput))
     #44 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(261): _PHPStan_781aefaf6\Symfony\Component\Console\Application->doRunCommand(Object(PHPStan\Command\WorkerCommand), Object(_PHPStan_781aefaf6\Symfony\Component\Console\Input\ArgvInput),
     Object(_PHPStan_781aefaf6\Symfony\Component\Console\Output\ConsoleOutput))
     #45 phar:///vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(157): _PHPStan_781aefaf6\Symfony\Component\Console\Application->doRun(Object(_PHPStan_781aefaf6\Symfony\Component\Console\Input\ArgvInput), Object(_PHPStan_781aefaf6\Symfony\Component\Console\Output\ConsoleOutput))
     #46 phar:///vendor/phpstan/phpstan/phpstan.phar/bin/phpstan(114): _PHPStan_781aefaf6\Symfony\Component\Console\Application->run()
     #47 phar:///vendor/phpstan/phpstan/phpstan.phar/bin/phpstan(115): {closure:phar:///vendor/phpstan/phpstan/phpstan.phar/bin/phpstan:14}()
     #48 /vendor/phpstan/phpstan/phpstan(8): require('phar:///Users/m...')
     #49 /vendor/bin/phpstan(119): include('/Users/moritz/P...')
     #50 {main}
```

So the actual error cause is the [`SquashedMigrationHelper`](https://github.com/larastan/larastan/blob/3.x/src/Properties/SquashedMigrationHelper.php#L68), which attempts to parse a squashed migration file to extract database table types. 

**This is obviously horrible error handling on the SQL Parser's part, but also makes me wonder why larastan uses a MySQL-only SQL parsing library?** We use postgres, and also have a project using sqlite; this seems like a really egregious issue, basically ruling out Larastan.

--- a/composer.json
+++ b/composer.json
@@ -17,7 +17,7 @@
     "require": {
         "php": "^8.2",
         "ext-json": "*",
-        "iamcal/sql-parser": "^0.5.0",
+        "iamcal/sql-parser": "^0.6.0",
         "illuminate/console": "^11.42.2 || ^12.0",
         "illuminate/container": "^11.42.2 || ^12.0",
         "illuminate/contracts": "^11.42.2 || ^12.0",

